src/main/java/org/jabref/preferences/FilePreferences.java

```diff
@@ -56,20 +57,20 @@ public String getUser() {
         return user.getValue();
     }
 
-    public Optional<Path> getFileDirectory() {
-        if (StringUtil.isBlank(mainFileDirectory.getValue())) {
-            return Optional.empty();
-        } else {
-            return Optional.of(Path.of(mainFileDirectory.getValue()));
-        }
+    public Optional<Path> getMainFileDirectory() {
+        return mainFileDirectory.getValue();
     }
 
-    public StringProperty mainFileDirectoryProperty() {
+    public OptionalObjectProperty<Path> mainFileDirectoryProperty() {
         return mainFileDirectory;
     }
 
     public void setMainFileDirectory(String mainFileDirectory) {
-        this.mainFileDirectory.set(mainFileDirectory);
+        if (StringUtil.isBlank(mainFileDirectory)) {
+            this.mainFileDirectory.setValue(Optional.empty());
```

Comment:
I don't like having this as an optional here and passing optional as parameters is mostly a code smell. I see no advantage over the current design.